src/components/Buildbaron/BuildBaron.test.tsx

```diff
@@ -0,0 +1,263 @@
+          failedTestCount: 0,
+        },
+      },
+    },
+  },
+  {
+    request: {
+      query: GET_USER,
+    },
+    result: {
+      data: {
+        userId: "mohamed.khelif",
+        displayName: "Mohamed Khelif",
+      },
+    },
+  },
+];
+
+it("The BuildBaron component renders without crashing.", () => {
+  const ContentWrapper = () => (
+    <MockedProvider mocks={mocks} addTypename={false}>
+      <BuildBaron
+        data={buildBaronQuery}
+        error={null}
+        taskId={taskId}
+        loading={false}
+      />
+    </MockedProvider>
+  );
+  const { queryByDataCy } = render(withRouter(ContentWrapper), {
+    route: `/task/${taskId}`,
+    path: "/task/:id",
+  });
+
+  expect(queryByDataCy("bb-content")).toBeInTheDocument();
+  expect(queryByDataCy("bb-error")).toBeNull();
+});
+
+it("Clicking on file a new ticket dispatches a banner.", async () => {
+  const ContentWrapper = () => (
+    <MockedProvider mocks={mocks} addTypename={false}>
+      <BuildBaron
+        data={buildBaronQuery}
+        error={null}
+        taskId={taskId}
+        loading={false}
+      />
+    </MockedProvider>
+  );
+  const { queryByDataCy, getByText } = render(withRouter(ContentWrapper), {
+    route: `/task/${taskId}`,
+    path: "/task/:id",
+  });
+  fireEvent.click(queryByDataCy("file-ticket-button"));
+  await expect(getByText("File Ticket")).toBeInTheDocument();
+  await fireEvent.click(getByText("File Ticket"));
+
+  waitFor(() =>
+    expect(
+      getByText("Ticket successfully created for this task")
+    ).toBeInTheDocument()
+  );
+});
+
+it("The correct JiraTicket rows are rendered in the component", () => {
+  const ContentWrapper = () => (
+    <MockedProvider mocks={mocks} addTypename={false}>
+      <BuildBaron
+        data={buildBaronQuery}
+        error={null}
+        taskId={taskId}
+        loading={false}
+      />
+    </MockedProvider>
+  );
+  const { queryAllByDataCy, queryByDataCy } = render(
+    withRouter(ContentWrapper),
+    {
+      route: `/task/${taskId}`,
+      path: "/task/:id",
+    }
+  );
+
+  expect(queryAllByDataCy("jira-ticket-row")).toHaveLength(3);
+
+  expect(queryByDataCy("EVG-12345")).toBeInTheDocument();
+  expect(queryByDataCy("EVG-12346")).toBeInTheDocument();
+  expect(queryByDataCy("EVG-12347")).toBeInTheDocument();
+
+  expect(queryByDataCy("EVG-12345-badge")).toHaveTextContent("Resolved");
+  expect(queryByDataCy("EVG-12345-metadata")).toHaveTextContent(
+    "Created: Sep 23, 2020 Updated: Sep 23, 2020 Unassigned"
+  );
+
+  expect(queryByDataCy("EVG-12346-badge")).toHaveTextContent("Closed");
+  expect(queryByDataCy("EVG-12346-metadata")).toHaveTextContent(
+    "Created: Sep 18, 2020 Updated: Sep 18, 2020 Assignee: Some Name"
+  );
+
+  expect(queryByDataCy("EVG-12347-badge")).toHaveTextContent("Open");
+  expect(queryByDataCy("EVG-12347-metadata")).toHaveTextContent(
+    "Created: Sep 18, 2020 Updated: Sep 18, 2020 Assignee: Backlog - Evergreen Team"
+  );
+});
```

Comment:
Would it be possible to capture this functionality in a test. To ensure the polling for a recently created ticket works. https://github.com/evergreen-ci/spruce/blob/5133e34d05a2cd50429103521fa3608cb41f4469/src/components/Buildbaron/BBCreatedTickets.tsx#L45-L50

Reply:
discussed over slack and decided not to do